### ISSUE TITLE
tensorflow timeline  /  gradients allreduce on cpu?

### ISSUE THREAD

**Issue by zymazloa** [uber/horovod#691]
my test is base on tensorflow/benchmarks，variable_update：horovod

tensorflow timeline : 

/job:localhost/replica:0/task:0/device:CPU:0 Compute
-------------------------------------------------------------------------
                      ||             | HorovodAllreduce |
                                     |           HorovodAllreduce               |
--------------------------------------------------------------------------

Title	                                HorovodAllreduce
Category	                        Op
User Friendly Category	other
Start	                                221.970 ms
Wall Duration	                238.885 ms
Args
  input0	"v0/tower_0/gradients/AddN_20/_2771"
  name	"v0/tower_0/HorovodAllreduce_v0_tower_0_gradients_AddN_20_0"
  op	        "HorovodAllreduce"

Events:
Incoming flow	v0/tower_0/gradients/AddN_20
Outgoing flow	v0/tower_0/gradients/AddN_20
Outgoing flow	v0/tower_0/HorovodAllreduce_v0_tower_0_gradients_AddN_20_0
Preceding events	3 events of various types
Following events	5 events of various types
All connected events	7 events of various types

does it mean hvd.allreduce work on cpu ?? or ??

INSTALL:   sudo ldconfig /usr/local/cuda-9.0/lib64 &&  HOROVOD_GPU_ALLREDUCE=NCCL sudo pip install --no-cache-dir horovod && sudo    ldconfig

USAGE:     

//loss
 with tf.device(/gpu:0):
       hvd.allgather(pre_logits)
       .......   
       hvd.allreduce(....)
       .......

//gradients
 with tf.device(/gpu:0):
     grads = [ hvd.allreduce(x) for x in grads] 

//run
CUDA_VISIBLE_DEVICES=1,2,3,4,5,6,7,8,9 \
mpirun -np 9 \
       -bind-to none -map-by slot \
       -mca btl_tcp_if_include ib0 \
python xx.py  \

               
(sorry  i can not upload supported files.......)

**Comment by alsrgv** [comment]
@zymazloa,

`HOROVOD_GPU_ALLREDUCE=NCCL sudo pip install --no-cache-dir horovod` could be causing a problem since `sudo` clears environment.

Can you try `sudo HOROVOD_GPU_ALLREDUCE=NCCL pip install --no-cache-dir horovod` instead?

**Comment by zymazloa** [comment]
@alsrgv    

it works !  thanks !
